packages/manager/src/views/WebPropertyEnvPage/components/EditAccess/EditMemberAccess.tsx

```diff
@@ -0,0 +1,206 @@
+    email: string
+  ) => {
+    const target = event.currentTarget as HTMLInputElement;
+    const temp = group.data;
+    const objIndex = temp.findIndex((e: GroupItem1) => e.email === email);
+    temp[objIndex][target.name] = checked;
+    setGroup({ data: temp });
+    if (checked) {
+      setAddAccess([...addAccess, target.name]);
+      setDeleteAccess(deleteAccess.filter((value) => value !== target.name));
+    } else {
+      setDeleteAccess([...deleteAccess, target.name]);
+      setAddAccess(addAccess.filter((value) => value !== target.name));
+    }
+  };
+  interface GroupData {
+    email: string;
+    name: string;
+    role: string;
+    [key: string]: unknown;
+  }
+  const handleSubmit = () => {
+    const addPerm = group.data.map(({ email, name, role, ...rand }: GroupData) => ({
+      email,
+      name,
+      actions: Object.keys(rand)
+    }));
+    const addData = { propertyIdentifier, permissionDetails: addPerm };
+    const deletePerm = group.data.map(({ email, name, role, ...rand }: GroupData) => ({
+      email,
+      name,
+      actions: Object.keys(rand).filter((value) => rand[value] === false)
+    }));
+    const deleteData = { propertyIdentifier, permissionDetails: deletePerm };
+    try {
+      if (addAccess.length) {
+        useAddPermission1
+          .mutateAsync({
+            ...addData
+          })
+          .then((res) => {
+            toast.success('Permission updated successfully');
+            return res;
+          });
+      }
+      if (deleteAccess.length) {
+        deleteMember
+          .mutateAsync({
+            ...deleteData
+          })
+          .then(() => {
+            toast.success('Permission updated successfully');
+          });
+      }
+      onClose();
+    } catch (err) {
+      // console.error(err);
+    }
+  };
+  return (
+    <div>
+      <Accordion asDefinitionList>
+        <AccordionItem>
+          <AccordionToggle
+            style={{ width: '100%' }}
+            onClick={() => {
+              onToggleAccordian(`def-list-toggle1_${editMemberName}`);
+            }}
+            isExpanded={expanded === `def-list-toggle1_${editMemberName}`}
+            id="def-list-toggle1_"
+          >
+            <Split>
+              <SplitItem style={{ display: 'flex', justifyContent: 'start' }} isFilled>
+                {editMemberName}
+              </SplitItem>
+              <SplitItem style={{ display: 'flex', justifyContent: 'end' }} isFilled>
+                {toPascalCase(group.data[0].role)}
+              </SplitItem>
+            </Split>
+          </AccordionToggle>
+          <AccordionContent
+            id={`def-list-expand1${editMemberName}`}
+            isHidden={expanded !== `def-list-toggle1_${editMemberName}`}
+          >
+            <TableComposable>
+              <Thead noWrap>
+                <Tr>
+                  <Th>{columnNames.NAME}</Th>
+                  <Th>{toPascalCase(columnNames.APIKEY_CREATION)}</Th>
```

Comment:
Why not just type is out the title 😅 if this is constant always

Reply:
going forward we may add more permissions and at that time we dont want to type in each and everyplace so I used one object, I will save it in one file to reuse it in all of them

Reply:
added ad todo